### PR TITLE
[bitnami/common] Add the ability to specify namespaces for affinity

### PR DESCRIPTION
**Pull request by klamas1**
### Description of the change

Added the ability to specify namespaces for affinity

### Benefits

You can create affinity and antiaffinity for workloads in different namespaces (I agree, this is a corner case, but still)

### Possible drawbacks

None
By default, the behavior of templates does not change

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [ ] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by klamas1** [comment]
@carrodher, do you have any questions about the changes?

File: bitnami/common/templates/_affinities.tpl [hunk 1]
```diff
@@ -60,13 +60,14 @@ Return a topologyKey definition
 
 {{/*
 Return a soft podAffinity/podAntiAffinity definition
-{{ include "common.affinities.pods.soft" (dict "component" "FOO" "customLabels" .Values.podLabels "extraMatchLabels" .Values.extraMatchLabels "topologyKey" "BAR" "extraPodAffinityTerms" .Values.extraPodAffinityTerms "context" $) -}}
+{{ include "common.affinities.pods.soft" (dict "component" "FOO" "customLabels" .Values.podLabels "extraMatchLabels" .Values.extraMatchLabels "topologyKey" "BAR" "extraPodAffinityTerms" .Values.extraPodAffinityTerms "namespaces" list "context" $) -}}
```
Comment (jotamartos) [inline bitnami/common/templates/_affinities.tpl:63]: As this is a comment with a usage example, let's include a list of namespaces

```suggestion
{{ include "common.affinities.pods.soft" (dict "component" "FOO" "customLabels" .Values.podLabels "extraMatchLabels" .Values.extraMatchLabels "topologyKey" "BAR" "extraPodAffinityTerms" .Values.extraPodAffinityTerms "namespaces" (list "namespace1" "namespace2") "context" $) -}}
```

File: bitnami/common/templates/_affinities.tpl [hunk 2]
```diff
@@ -77,6 +78,13 @@ preferredDuringSchedulingIgnoredDuringExecution:
           {{- range $key, $value := $extraMatchLabels }}
           {{ $key }}: {{ $value | quote }}
           {{- end }}
+      {{- if $namespaces }}
+      namespaces:
+        - {{ .context.Release.Namespace }}
+        {{- with $namespaces }}
+        {{ include "common.tplvalues.render" (dict "value" . "context" $) | nindent 8 }}
+        {{- end }}
+      {{- end }}
```
Comment (jotamartos) [inline bitnami/common/templates/_affinities.tpl:87]: As you are adding the Release.Namespace by default, what about if we name the variable "extraNamespaces"?

```suggestion
      {{- if $extraNamespaces }}
      namespaces:
        - {{ .context.Release.Namespace }}
        {{- with $extraNamespaces }}
        {{ include "common.tplvalues.render" (dict "value" . "context" $) | nindent 8 }}
        {{- end }}
      {{- end }}
```

**Review by jotamartos — CHANGES_REQUESTED** [review]
Thanks for contributing and sorry for the late review. Could you please take a look at my comments?

File: bitnami/common/templates/_affinities.tpl (hunk 2 above)
Comment (klamas1) [reply bitnami/common/templates/_affinities.tpl:87]: I agree, this is a clearer example

Comment (klamas1) [reply bitnami/common/templates/_affinities.tpl]: @jotamartos thank you for review,
I made the suggested changes

**Review by jotamartos — APPROVED** [review]

**Pull request merged**